internal/resources/iampolicy/resource_iam_policy.go

```diff
@@ -0,0 +1,424 @@
+/*
+Copyright © 2022 VMware, Inc. All Rights Reserved.
+SPDX-License-Identifier: MPL-2.0
+*/
+
+package iampolicy
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+
+	"github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/authctx"
+	"github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/helper"
+	clustermodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/cluster"
+	clustergroupmodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/clustergroup"
+	iammodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/iam_policy"
+	clusteriammodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/iam_policy/cluster"
+	clustergroupiammodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/iam_policy/clustergroup"
+	namespaceiammodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/iam_policy/namespace"
+	organizationiammodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/iam_policy/organization"
+	workspaceiammodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/iam_policy/workspace"
+	namespacemodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/namespace"
+	organizationmodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/organization"
+	workspacemodel "github.com/vmware-tanzu/terraform-provider-tanzu-mission-control/internal/models/workspace"
+)
+
+func ResourceIAMPolicy() *schema.Resource {
+	return &schema.Resource{
+		ReadContext:   dataSourceIAMPolicyRead,
+		CreateContext: resourceIAMPolicyCreate,
+		//UpdateContext: resourceIAMPolicyUpdate,
+		DeleteContext: resourceIAMPolicyDelete,
+		Schema:        iamPolicySchema,
+	}
+}
+
+var iamPolicySchema = map[string]*schema.Schema{
+	scopeKey:       scopeSchema,
+	roleBindingKey: roleBinding,
+}
+
+var scopeSchema = &schema.Schema{
```

Comment:
Can we split these schema definitions and their respective construct/expand and latten functions into separate packages or files?

Reply:
Done